packages/compass-e2e-tests/tests/oidc.test.ts

```diff
@@ -0,0 +1,168 @@
+import type { CompassBrowser } from '../helpers/compass-browser';
+import { beforeTests, afterTests } from '../helpers/compass';
+import type { Compass } from '../helpers/compass';
+import type { OIDCMockProviderConfig } from '@mongodb-js/oidc-mock-provider';
+import { OIDCMockProvider } from '@mongodb-js/oidc-mock-provider';
+import type { ChildProcess } from 'child_process';
+import { spawn } from 'child_process';
+import path from 'path';
+import os from 'os';
+import { promises as fs } from 'fs';
+import { getDownloadURL } from 'mongodb-download-url';
+import tar from 'tar';
+import { promisify } from 'util';
+import type { Readable } from 'stream';
+import { pipeline, PassThrough } from 'stream';
+import { createInterface as readline } from 'readline';
+import https from 'https';
+import { once } from 'events';
+import { expect } from 'chai';
+
+describe('OIDC integration', function () {
+  let compass: Compass;
+  let browser: CompassBrowser;
+
+  let getTokenPayload: typeof oidcMockProviderConfig.getTokenPayload;
+  let oidcMockProviderConfig: OIDCMockProviderConfig;
+  let oidcMockProvider: OIDCMockProvider;
+
+  let i = 0;
+  let tmpdir: string;
+  let server: ChildProcess;
+  let serverExit: Promise<unknown>;
+  let connectionString: string;
+
+  before(async function () {
+    // TODO(MONGOSH-1306): Get rid of all the setup code to download mongod here... :(
+    if (process.platform !== 'linux') {
+      return this.skip();
```

Comment:
Copying some of the comment over from https://github.com/mongodb-js/mongosh/pull/1458/files#diff-00a4109f7ba4dab51e247b01caab1dc7ceff5e3f32b8e5a03cab89eedd14d8d8R33
```suggestion
      // OIDC is only supported on Linux in the 7.0+ enterprise server.
      return this.skip();
```